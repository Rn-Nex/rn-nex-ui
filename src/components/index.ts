export * from './Accordion';
export * from './ActivityIndicator';
export * from './Avatar';
export * from './Badge';
export * from './Box';
export * from './Button';
export * from './Card';
export * from './CheckBox';
export * from './Chip';
export * from './Dialog';
export * from './Divider';
export * from './DropDown';
export * from './Image';
export * from './List';
export * from './Pagination';
export * from './Portal';
export * from './Radio';
export * from './Snackbar';
export * from './Stepper';
export * from './Switch';
export * from './TextField';
export * from './Transitions';
export * from './Typography';
