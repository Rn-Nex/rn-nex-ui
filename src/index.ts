export * from './components';
export * from './libraries';
export * from './utils';
