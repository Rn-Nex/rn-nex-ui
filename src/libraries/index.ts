export * from './themes/v2';
export * from './style';
