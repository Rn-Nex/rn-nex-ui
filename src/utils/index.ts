export * from './styleGenerator';
export * from './randomId';
export * from './utils';
export * from './dimensions';
