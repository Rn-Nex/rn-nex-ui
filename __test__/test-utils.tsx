import React from 'react';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from '../src';

const AllTheProviders: React.FC<{ children: React.ReactNode }> = ({ children }) => {
  return <ThemeProvider>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';
export { customRender as render };
