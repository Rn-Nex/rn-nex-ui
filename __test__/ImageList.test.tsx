import React from 'react';
import { render } from './test-utils';
import { ImageList } from '../src';
import { ScrollView, View } from 'react-native';

describe('ImageList Component', () => {
  const mockImageListTestId = 'image-list-test-id';
  const mockChildTestId = 'child-test-id';

  const mockRef = React.createRef<ScrollView>();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly with default props', () => {
    const { toJSON } = render(<ImageList />);
    expect(toJSON).toMatchSnapshot();
  });

  it('should forward ref correctly', () => {
    render(<ImageList ref={mockRef} />);
    expect(mockRef.current).toBeInstanceOf(ScrollView);
    expect(mockRef.current).not.toBeNull();
  });

  it('should render the child component correctly', () => {
    const { getByTestId } = render(
      <ImageList>
        <View testID={mockChildTestId} />
      </ImageList>,
    );

    const child = getByTestId(mockChildTestId);
    expect(child).toBeDefined();
  });
});
