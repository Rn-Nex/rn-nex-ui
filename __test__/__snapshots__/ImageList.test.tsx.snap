// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageList Component should render correctly with default props 1`] = `[Function]`;

exports[`ImageListItem Component should render correctly 1`] = `[Function]`;
